packages/grid/src/Grid.tsx

```diff
@@ -1,81 +1,313 @@
 import React, {
-  useRef, useState, useEffect, useCallback,
+  useState,
+  useRef,
+  useEffect,
+  useCallback,
+  MutableRefObject,
+  useContext,
 } from 'react';
+import { ThemeContext } from 'styled-components';
 import {
-  IColumn, ITable, ITableProps, IRowData, ITableTheme, IRow,
-} from './interfaces';
-import { Row } from './Row';
+  Grid as VirtualisedGrid,
+  CellMeasurer,
+  CellMeasurerCache,
+  ScrollPosition,
+  GridCellProps,
+} from 'react-virtualized';
+import { setIn } from 'utilitify';
 import {
-  ContentWrapper, HeadWrapper, TableHead, TableStyled, TableWrapper, HeaderStyled,
+  Body,
+  BodyCell,
+  BodyCellOffset,
+  BodyCellContent,
+  ExpandButtonWrapper,
+  Wrapper,
+  TotalBlock,
+  TotalCellContent,
+  TotalCell,
 } from './styled';
-import { gridTheme } from './utils/get-styles';
-
-
-export const Grid: React.FC<ITable> = React.memo<ITableProps>(({
-  rows,
-  columns,
-  theme = {},
-  multiline = false,
-}) => {
-  const [selectedRows, changeSelectedRows] = useState<IRow[] | []>([]);
-  const tableWrapperElement = React.createRef<HTMLDivElement>();
-  const [tableScroll, setTableScroll] = useState(0);
-  const themeRef = useRef(gridTheme<ITableTheme>(theme));
-
-  const handleScroll = (): void => {
-    const scrolledDiv = tableWrapperElement.current;
-    if (scrolledDiv) {
-      setTableScroll(scrolledDiv.scrollLeft);
+import { AddIcon, RemoveIcon } from './icons';
+import {
+  guid, addOrDeleteItemFromArray, gridTheme, deletePropsFromObjects,
+} from './utils';
+
+import { HeaderWrapper } from './HeaderWrapper';
+import {
+  IItem, IColumn, IGrig, IMappedItem, IGridTheme,
+} from './interfaces';
+
+
+const Grid = ({
+  items = [],
+  columns = [],
+  width = 0,
+  height = 0,
+  isDisableSelect = false,
+  isMultiSelect = false,
+  onChangeColumns = () => {},
+  totals,
+  theme,
+  onSelect = () => {},
+  shouldMovingColumns = true,
+  shouldChangeColumnsWidth = true,
+}: IGrig) => {
+  const contextTheme = useContext(ThemeContext);
+  const themeRef = useRef(gridTheme<IGridTheme>(theme || contextTheme));
+  const [mappedColumns, changeMappedColumns] = useState<IColumn[]>(columns);
```

Comment:
I think it is better to use **set** instead of **change** (**set**MappedColumns). 
In all examples of using useState, people use set, so it is more habitually. And **set** is shorter)